app/helpers/distribution_helper.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module DistributionHelper
+  VALUE = '_'
+
+  # Consider these as private methods that are (and can only be) tested in their own
+  # context, hence there is no specific test suite for this file.
+  def usable_questions
+    @questionnaire_content = questionnaire.content
+    range_questions(@questionnaire_content) + other_questions(@questionnaire_content)
+  end
+
+  def range_questions(questionnaire_content)
+    rg_instance = RangeGenerator.new
+    questionnaire_content
+      .select { |question| question[:type] == :range }
+      .map do |question|
+      rg_instance.send(:range_slider_minmax, question).merge(id: question[:id].to_s,
+                                                             type: question[:type],
+                                                             combines_with: question[:combines_with],
+                                                             step: (question[:step] || 1))
+    end
+  end
+
+  def other_questions(questionnaire_content)
+    questionnaire_content
+      .select { |question| %i[number radio likert dropdown].include?(question[:type]) }
+      .map do |question|
+      { id: question[:id].to_s, type: question[:type], combines_with: question[:combines_with] }
+    end
+  end
+
+  def initialize_question(question, value, distribution)
+    qid = question[:id]
+    return if distribution[qid].present? && question[:type] == :range
+
+    distribution[qid] ||= {}
+    distribution[qid][value] ||= { VALUE => 0 } unless question[:type] == :range
+    return unless question[:type] == :range
+
+    pos = question[:min]
+    while pos <= question[:max]
+      distribution[qid][pos.to_s] = { VALUE => 0 }
+      pos += question[:step]
+    end
```

Comment:
```suggestion
    (question[:min]..question[:max]).step(question[:step]) { |pos| distribution[qid][pos.to_s] = { VALUE => 0 } }
```

Reply:
(please test this)

Reply:
Ok i will write a spec for this method and then try your implementation and see if it still passes.

Reply:
Yep it seems to work identical. Fixed.

Reply:
Nice.